### ISSUE TITLE
[TFB-1449] Add basic git-review setup/usage to Contributors

### ISSUE THREAD

**Issue by pono** [tungstenfabric/docs#38]
Here are notes I took while setting up git-review and sending my first patch (for https://jira.tungsten.io/browse/TFB-1412 ) They need to be cleaned up, converted to .rst, and added to the Contributors directory.

- Confirm you have a Jira ticket for the work (required for commit)
- Confirm you have a signed CLA?
  - Gerrit > Settings > Agreements > New Contributor Agreement OR
  - https://contributor.lfcla.com/#/login
  - Should be directed to this first time you try to do a git-review?
- Clone the repo (if you don’t have it yet)
  - Log into gerrit.tungsten.io
  - Click Browse > Repositories
  - Select the repo you want
  - git clone "ssh://vmbrasseur@gerrit.tungsten.io:29418/docs"
- Set up SSH access for Gerrit (required for the scp in the next step)
  - Log into gerrit.tungsten.io (if you’re not already)
  - Click the ‘Settings’ gear in the upper right next to your name
  - Click ‘SSH Keys’
  - Add your *public* SSH key (.pub)
  - Test: ssh -p 29418 vmbrasseur@gerrit.tungsten.io gerrit version
  - Should be `gerrit version 2.16.8` or higher
- Set up git-review
  - References
    - section 4.2: https://github.com/tungstenfabric/docs/blob/master/Contributor/GettingStarted/getting-started-with-tungstenfabric-development.md
    - https://docs.openstack.org/infra/git-review/
    - https://www.mediawiki.org/wiki/Gerrit/git-review
  - pip install —user git-review
  - Update path to include git-review ($HOME/Library/Python/2.7/bin on macOS; ~/.local/bin on Ubuntu thanks to that --user in the pip command)
  - export PATH=$HOME/Library/Python/2.7/bin:$PATH <— .bash_profile
  - cd $repo
- Set git config for the repo
  - git config user.email vmb@juniper.net￼
  - gitdir=$(git rev-parse --git-dir);
  - scp -p -P 29418 vmbrasseur@gerrit.tungsten.io:hooks/commit-msg ${gitdir}/hooks (necessary? cloned w/hooks above)
  - git remote add gerrit ssh://vmbrasseur@gerrit.tungsten.io:29418/docs
  - git review -s (tests the setup; success == no messages)
- Make necessary changes
- Commit your  changes
  - Use `commit -s` so you’ll get the signed-off-by, which is required by git review
  - If you forget to sign a commit, run `git commit --amend -s -c <SHA>` on it
- Run `git review` to get the patch into Gerrit
  - git review

Success looks something like this:
`Θυια:docs vmb$ git review

remote:

remote: Processing changes: }}{{new: 1 ()

remote: Processing changes: new: 1 (|)

remote: Processing changes: new: 1 

remote: Processing changes: new: 1 

remote: Processing changes: new: 1 ()

remote: Processing changes: refs: 1, new: 1 ()

remote: Processing changes: refs: 1, new: 1, done

remote:

remote: SUCCESS

remote:

remote: New Changes:

remote:   https://gerrit.tungsten.io/r/c/docs/+/24 Reorg docs directories

To ssh://gerrit.tungsten.io:29418/docs 

* [new branch]      HEAD -> refs/for/master
`

**Comment by pono** [comment]
Notes from Will:
After much trial and error, I was able to get a patch submitted.
 
My commit message looks like this:
 
$ git log
commit a474020b9c42f2d0526793a9534f8f9e38107b77 (HEAD -> swill)
Author: swill <williamstevens@gmail.com>
Date:   Fri Sep 13 15:02:36 2019 -0400

    Updated the docs to be Read The Docs compatible
   
    Jira Ticket: https://jira.tungsten.io/browse/TFB-1481
    Closes: TFB-1481
    Issue-ID: TFB-1481
   
    Change-Id: I7c211b4799989f7563fe2f752528698c897e7d03
    Signed-off-by: swill <williamstevens@gmail.com>
 
It turns out that `Issue-ID:` is CASE SENSITIVE, so it has to be written EXACTLY as specified with case, etc...
 
It was pushed with the following command:
 
$ git push gerrit HEAD:refs/for/master%topic=swill-read-the-docs
 
I followed Vicky's instructions for the most part: https://jira.tungsten.io/browse/TFB-1449
 
I didn't do these steps because I didn't understand what it was trying to make me do, so I skipped these steps:
 
$ gitdir=$(git rev-parse --git-dir);
$ scp -p -P 29418 vmbrasseur@gerrit.tungsten.io:hooks/commit-msg ${gitdir}/hooks (necessary? cloned w/hooks above)
 
I used this extensively trying to get a format which was accepted:
 
$ git commit --amend -s -c 7359a37704ac5fd507ee4055c977413afc1c6281
 
Hopefully this helps.  Let me know if you still have issues and I will see if I can help.
